realm/realm-jni/src/io_realm_internal_UncheckedRow.cpp

```diff
@@ -217,7 +217,7 @@ JNIEXPORT jlong JNICALL Java_io_realm_internal_UncheckedRow_nativeGetLinkView
     if (!ROW_VALID(env, ROW(nativeRowPtr)))
         return 0;
 
-    LinkView* link_view_ptr = LangBindHelper::get_linklist_ptr( *ROW( nativeRowPtr ), S( columnIndex) );
+    LinkViewRef* link_view_ptr = const_cast<LinkViewRef*>(&(LangBindHelper::get_linklist_ptr(*ROW(nativeRowPtr), S(columnIndex))));
     return reinterpret_cast<jlong>(link_view_ptr);
```

Comment:
I am not quite sure what you mean: no strings are involved there. With core 0.100.0, `LangBindHelper::get_linklist_ptr()` has been changed to return a `const LinkViewRef` (a `shared_ptr<LinkView*>`. As we cannot use a `shared_ptr` in Java, I have to store a pointer to `shared_ptr` (returned as a `jlong`). And I need to get rid of the constness too.

That gives us a double indirection which you can see in all the `LinkView` methods.